scripts/semconv/templates/SemanticAttributes.ts.j2

```diff
@@ -24,9 +37,74 @@
 {%- macro lowerFirst(text) -%}
   {{ text[0]|lower}}{{text[1:] }}
 {%- endmacro %}
+{%- macro format_value(type, value, depth = 0, orgValue = value) -%}
```

Comment:
This macro is pretty much where all of the dynamic string concatenation (of the defined constants and dotConstants) occurs.

It's a simple left to right recursive parser, once it can't find any more matches it stops and just dumps the remaining string value as a quoted constant.